src/snowflake/snowpark/_internal/ast/utils.py

```diff
@@ -1494,3 +1495,28 @@ def base64_lines_to_textproto(base64_str: str) -> str:
 def textproto_to_request(textproto_str: str) -> proto.Request:
     request = Parse(textproto_str, proto.Request())
     return request
+
+
+def clear_line_no_in_ast(ast: Any) -> None:
+    """Clear any 'src' information in the statement body."""
+    if isinstance(ast, Iterable) and not isinstance(ast, str):
+        for c in ast:
+            clear_line_no_in_ast(c)
+    elif hasattr(ast, "DESCRIPTOR"):
+        if hasattr(ast, "src"):
+            ast.ClearField("src")  # type: ignore[union-attr]
+
+        for f in ast.DESCRIPTOR.fields:
+            c = getattr(ast, f.name)
+            if (isinstance(c, Iterable) and not isinstance(c, str) and len(c) > 0) or (  # type: ignore[arg-type]
+                isinstance(c, Message) and c.ByteSize() > 0
+            ):
+                clear_line_no_in_ast(c)
+
+
+def clear_line_no_in_request(request: proto.Request) -> None:
+    """There are inconsistencies in the frame_info.line_no depending on the python version, this seems to be due to
```

Comment:
can we update the comment based on offline convo? I think it's also fine to remove any comment here mentioning the python version and simply describe what the function does. The whole intention on for which version we call this function should be commented where the function is called.